server/core/src/main/scala/org.apache.daffodil.tdml/TDML.scala

```diff
@@ -17,232 +17,10 @@
-  // There is a suiteName attribute in the root element of the document. This is set to tdmlName
-  def generate(
-      infosetPath: String,
-      schemaPath: String,
-      dataPath: String,
-      tdmlName: String,
-      tdmlDescription: String,
-      tdmlPath: String
-  ): Unit = {
-    val factory = new ObjectFactory()
-
-    val testSuite = factory.createTestSuite()
-    testSuite.setSuiteName(tdmlName)
-    testSuite.setDefaultRoundTrip(RoundTripType.ONE_PASS)
-    testSuite
-      .getTutorialOrParserTestCaseOrDefineSchema()
-      .add(createTestCase(infosetPath, schemaPath, dataPath, tdmlName, tdmlDescription))
-
-    val marshaller = JAXBContext.newInstance(classOf[TestSuite]).createMarshaller()
-    marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true)
-    marshaller.marshal(testSuite, new java.io.File(tdmlPath))
-  }
-
-  // Append a new test case to an existing TDML file.
-  // Paths given to this function should be absolute as they will be converted to relative paths
-  //
-  // infosetPath:     Path to the infoset
-  // schemaPath:      Path to the DFDL Schema
-  // dataPath:        Path to the data file
-  // tdmlName:        Name of the DFDL operation
-  // tdmlDescription: Description for the DFDL operation
-  // tdmlPath:        Path to the TDML file
-  def append(
-      infosetPath: Path,
-      schemaPath: Path,
-      dataPath: Path,
-      tdmlName: String,
-      tdmlDescription: String,
-      tdmlPath: String
-  ): Unit =
-    append(
-      convertToRelativePath(infosetPath, tdmlPath),
-      convertToRelativePath(schemaPath, tdmlPath),
-      convertToRelativePath(dataPath, tdmlPath),
-      tdmlName,
-      tdmlDescription,
-      tdmlPath
-    )
-
-  // Append a new test case to an existing TDML file.
-  // Paths given to this function should be relative as it should be expected for the TDML files to be shared on the mailing list
-  //
-  // infosetPath:     Path to the infoset
-  // schemaPath:      Path to the DFDL Schema
-  // dataPath:        Path to the data file
-  // tdmlName:        Name of the DFDL operation
-  // tdmlDescription: Description for the DFDL operation
-  // tdmlPath:        Path to the TDML file
-  def append(
-      infosetPath: String,
-      schemaPath: String,
-      dataPath: String,
-      tdmlName: String,
-      tdmlDescription: String,
-      tdmlPath: String
-  ): Unit = {
-
-    val testSuite = JAXBContext
-      .newInstance(classOf[TestSuite])
-      .createUnmarshaller()
-      .unmarshal(new File(tdmlPath))
-      .asInstanceOf[TestSuite]
-
-    testSuite
-      .getTutorialOrParserTestCaseOrDefineSchema()
-      .add(createTestCase(infosetPath, schemaPath, dataPath, tdmlName, tdmlDescription))
-
-    val marshaller = JAXBContext.newInstance(classOf[TestSuite]).createMarshaller()
-    marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true)
-    marshaller.marshal(testSuite, new java.io.File(tdmlPath))
-  }
+  case class Execute(schema: Path, data: Path)
```

Comment:
I removed the other actions to make the initial work easier. `Execute` only reads an XML file and translates it into a debugging session.